.github/workflows/all_diffs.yml

```
@@ -0,0 +1,166 @@
+name: Run Text and Image Diff Checks
+on:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0'
+  workflow_dispatch:
+
+concurrency:
+  group: get-diff-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  get-dev-build:
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          ref: master
+
+      - name: Get Workflow run ID
+        id: workflow_run_id
+        uses: actions/github-script@v6
+        with:
+          result-encoding: string
+          script: |
+            const workflowRuns = await github.paginate(github.rest.actions.listWorkflowRuns.endpoint.merge({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: 'build-branch-dev.yml',
+              branch: 'master',
+              event: 'schedule',
+              status: 'success'
+            }));
+            const workflowRunsSorted = workflowRuns.sort((a, b) => b.run_number - a.run_number);
+
+            return workflowRunsSorted[0].id
+
+    outputs:
+      workflow_id: ${{ steps.workflow_run_id.outputs.result }}
+
+  get-master-build:
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          ref: master
+
+      - name: Get Workflow run ID
+        id: workflow_run_id
+        uses: actions/github-script@v6
+        with:
+          result-encoding: string
+          script: |
+            const workflowRuns = await github.paginate(github.rest.actions.listWorkflowRuns.endpoint.merge({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: 'build-branch-master.yml',
+              branch: 'master',
+              event: 'schedule',
+              status: 'success'
+            }));
+            const workflowRunsSorted = workflowRuns.sort((a, b) => b.run_number - a.run_number);
+
+            return workflowRunsSorted[0].id
+
+    outputs:
+      workflow_id: ${{ steps.workflow_run_id.outputs.result }}
+
+  check-diffs:
+    runs-on: ubuntu-latest
+    needs: [get-dev-build, get-master-build]
+
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          ref: diff_action
+
+      - name: Create Build Folders
+        run: |
+          mkdir dev-build
+          mkdir master-build
+
+      - name: Download Dev Artifact
+        uses: XanaduAI/cloud-actions/download-github-workflow-artifact@main
+        with:
+          workflow_run_id: ${{ needs.get-dev-build.outputs.workflow_id }}
+          artifact_name_regex: html.zip
+          github_token: ${{ github.token }}
+          artifact_download_dir: ${{ github.workspace }}/dev-build
+
+      - name: Unpack Dev Artifact
+        run: |
+          cd dev-build
+          FILE='html.zip.zip'
+          if [ -f "$FILE" ]; then
+            echo "$FILE exists."
+            unzip html.zip
+          fi
```

Let's also add a rm step at the bottom so we know the next step is downloading the correct new file, since both have the same name.
```suggestion
      - name: Unpack Dev Artifact
        run: |
          cd dev-build
          FILE='html.zip.zip'
          if [ -f "$FILE" ]; then
            echo "$FILE exists."
            unzip html.zip
          fi
          rm $FILE
```